From 368da4f4f5e4d05aecf149a5773d8d0b4f28ff92 Mon Sep 17 00:00:00 2001
From: Julien Rische <jrische@redhat.com>
Date: Tue, 18 Nov 2025 11:17:27 +0100
Subject: [PATCH] ipatests: fix kdcproxy tests against AD

In recent kdcproxy commit 1773f28eeea72ec6efcd433d3b66595c44d1253f, the
use of DNS discovery has been restricted to declared realms in order to
mitigate CVE-2025-59088. The present commit update the
TestHttpKdcProxy::test_ad_user_login_on_client_with_kdcproxy test to
take this change into account.

Signed-off-by: Julien Rische <jrische@redhat.com>
Reviewed-By: Florence Blanc-Renaud <flo@redhat.com>
---
 ipatests/test_integration/test_http_kdc_proxy.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ipatests/test_integration/test_http_kdc_proxy.py b/ipatests/test_integration/test_http_kdc_proxy.py
index b5906bc4bf8de7000d2b8ec582cbacf6e2e05be0..d48a5dd953d58a002fe7247d98e27c82a72b6626 100644
--- a/ipatests/test_integration/test_http_kdc_proxy.py
+++ b/ipatests/test_integration/test_http_kdc_proxy.py
@@ -131,8 +131,8 @@ class TestHttpKdcProxy(IntegrationTest):
         with tasks.remote_ini_file(self.master, paths.KDCPROXY_CONFIG) as conf:
             conf.set('global', 'use_dns', 'true')
             conf.set('global', 'configs', 'mit')
+            conf.add_section(self.ad.domain.realm)
             if use_tcp:
-                conf.add_section(self.ad.domain.realm)
                 conf.set(self.ad.domain.realm, 'kerberos',
                          'kerberos+tcp://{}:88'.format(self.ad.hostname))
                 conf.set(self.ad.domain.realm, 'kpasswd',
-- 
2.52.0

